File: examples/postoffice/third/kid.rb
require 'dramatis/actor'

require 'mangler'

class Kid

  Dramatis::Actor.acts_as self
  
  def initialize name, next_kid = nil
    @name = name
    @next = next_kid
  end

  def to_s
    @name
  end

  def whisper what
    mangle( @next.nil? && @next.whisper( mangle( what )  ) || what )
  end

  def mangle what
    Mangler.mangle what
  end

end

File: examples/telephone/first/kid.py
import dramatis

from  mangler import Mangler

class Kid( dramatis.Actor ):
  
  def __init__(self, name, next_kid = None):
      super(Kid,self).__init__()
      self._name = name
      self._next = next_kid

  def __str__(self):
      return self._name
  
  def whisper(self, what):
      what = self.mangle( what )
      if self._next:
          self._next.whisper( what )
      else:
          print "%s: %s" % ( self, what )

  def mangle(self, what):
      return Mangler.mangle( what )
